clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -5985,10 +5985,14 @@ RValue CodeGenFunction::EmitBuiltinExpr(const GlobalDecl GD, unsigned BuiltinID,
       llvm::Value *Block =
           Builder.CreatePointerCast(Info.BlockArg, GenericVoidPtrTy);
 
-      AttrBuilder B(Builder.getContext());
-      B.addByValAttr(NDRangeL.getAddress().getElementType());
-      llvm::AttributeList ByValAttrSet =
-          llvm::AttributeList::get(CGM.getModule().getContext(), 3U, B);
+      llvm::AttributeList ByValAttrSet;
+      // AMDGPU doesn't use byval for struct argument.
```

Comment:
Should query the ABInfo for what to do for the signature

Reply:
In fact, the other `__enqueue_kernel*` functions don't have this at all. I removed the code directly to align with others. It might be quite complicated to try to get `ABIArgInfo` here.